src/components/Login.js

```diff
@@ -2,10 +2,30 @@ import React from 'react'
 import firebase from '../firebase_setup/firebase'
 
 class Login extends React.Component {
-  login() {
-    const email = document.querySelector('#email').value
-    const password = document.querySelector('#password').value
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      email: '',
+      password: '',
+    }
+  }
+  // e is the event, which in this case is change, target is the element that triggered the event,
+  // which in this case is the input, and value is the value of the input element
+  onChange = (e) => {
+    // console.log(e.target.id)
+    if (e.target.id === 'email') this.setState({ email: e.target.value })
+    if (e.currentTarget.id === 'password')
+      this.setState({ password: e.target.value })
+  }
 
+  //handles login using firebase.auth.signInWithEmailandPassword
+  login() {
```

Comment:
you've defined `onChange` as an arrow function, and `login` as a class method. what do you think is the difference between the two? what criteria should we be using to decide how we define our functions?